### ISSUE TITLE
Bug with transactions history when passing `productType` and probably any other array

### ISSUE THREAD

**Issue by neoighodaro** [readdle/app-store-server-api#24]
If you pass the `productType` parameter to the `getTransactionHistory` or `getTransactionHistoryV2` it fails with an exception:

```php
$history = AppStoreServerApi::getTransactionHistory($id, [
    'productId' => array_values(Config::array('appstore.product_id')),
    'productType' => [
        GetTransactionHistoryQueryParams::PRODUCT_TYPE__AUTO_RENEWABLE,
    ],
    'sort' => GetTransactionHistoryQueryParams::SORT__DESCENDING,
    'revoked' => false,
]);
``` 

Exception:

```
[2024-10-12 10:48:35] local.ERROR: [Readdle\AppStoreServerAPI\RequestQueryParams\GetTransactionHistoryQueryParams] Parameter "productType" has wrong value array (
  0 => 'AUTO_RENEWABLE',
) {"exception":"[object] (UnexpectedValueException(code: 0): [Readdle\\AppStoreServerAPI\\RequestQueryParams\\GetTransactionHistoryQueryParams] Parameter \"productType\" has wrong value array (
  0 => 'AUTO_RENEWABLE',
) at /[...]/vendor/readdle/app-store-server-api/src/Request/AbstractRequestParamsBag.php:109)
```

After some digging through, it turns out the offending code is here:

```php
protected function isValueMatchingPropValues($value, array $propValues): bool
{
    if (is_scalar($value)) {
        return in_array($value, $propValues);
    }

    if (is_array($value)) {
        return (bool) array_filter($value, fn ($v) => ! in_array($v, $propValues));
    }

    return false;
}
```

The array_filter function is filtering out elements that do not exist in `$propValues`, and then it’s casting the result to a boolean. 

So if any element in `$value` is not found in `$propValues`, the result will be `true`, which is then negated with `!`. As a result, when I pass, for example, `["A"]` and `["A", "B", "C]`, the function returns `false` because `array_filter` doesn’t return anything that isn’t found in `$propValues`.

<img width="1479" alt="Screenshot 2024-10-12 at 12 53 19 PM" src="https://github.com/user-attachments/assets/155da178-9007-4b3a-97ed-952187251190">

Not sure that's the expected behavior.

This could fix that.

<img width="1479" alt="Screenshot 2024-10-12 at 12 55 39 PM" src="https://github.com/user-attachments/assets/aa3238ac-b2da-4b66-806d-a9c6450929e0">

**Comment by neoighodaro** [comment]
There's also an issue passing "excludeRevoke" as `true`

**Comment by pkotets** [comment]
Hello,

Thanks so much for such a detailed report.
Indeed, there was a mistake in `isValueMatchingPropValues()` method, double negation should be just a negation. Probably typo. I saw your PR, but in order to keep diffs as simple as possible, I just removed extra negation.

Also, speaking of "excludeRevoked", I never tested this functionality with bool variables, so there was a bug. I added additional `elseif` branch for this purpose and tested it with the list of params you provided, seems everything is fine now.

New version 3.10.1 has just been released, let me know if it fixes everything you mentioned.
